bool is_detect
---
bool is_ok